### ISSUE TITLE
NPM scripts explorer doesn't show scripts with defined tasks when in a multi-root workspace

### ISSUE THREAD

**Issue by eamodio** [Microsoft/vscode#54085]
When you are using a normal folder everything works fine -- all scripts show up regardless of configured tasks. But if you open the same folder in a multi-root workspace, then only scripts without configured task show up in the explorer.

I really thought I was going nuts for a bit there 😄 

This issue is related to this https://github.com/Microsoft/vscode/issues/48831 and https://github.com/Microsoft/vscode/issues/49197

Does this issue occur when all extensions are disabled?: Yes/No

**Comment by dbaeumer** [comment]
@egamma starting with you. Please assign back if there is something that needs to be done on the API side.

**Comment by eamodio** [comment]
FYI, I've noticed some other details around this. It seems like the first folder in a workspace does work, but all the subsequent folders only show npm scripts that don't have configured tasks.

**Comment by egamma** [comment]
Is the repository available so that I can reproduce it? I fail to reproduce it:
- git clone https://github.com/Microsoft/vscode-tslint.git
- open the workspace file vscode-tslint.code-workspace
- expand the npm script explorer

The script tslint-tests/lint shows up and it is a [customized task](https://github.com/Microsoft/vscode-tslint/blob/master/tslint-tests/.vscode/tasks.json).

![image](https://user-images.githubusercontent.com/172399/42733785-f78aac54-8837-11e8-873b-41ddd47c7029.png)

**Comment by egamma** [comment]
@eamodio pls see this comment https://github.com/Microsoft/vscode/issues/54307#issuecomment-405269079

**Comment by eamodio** [comment]
I created a workspace:
```json
{
	"folders": [
		{
			"path": "vscode-gitlens"
		},
		{
			"path": "vscode-remotehub"
		}
	],
	"settings": {}
}
```

Where [GitLens](https://github.com/eamodio/vscode-gitlens) and [RemoteHub](https://github.com/eamodio/vscode-remotehub) are cloned into the same root directory.

And depending on which folder is first determines which tasks go missing.

This is with GitLens first:
![image](https://user-images.githubusercontent.com/641685/42768529-c7a5b9c8-88ed-11e8-8789-edaf14af9c69.png)

This is with RemoteHub first:
![image](https://user-images.githubusercontent.com/641685/42768559-dbf12480-88ed-11e8-82a8-3393398ec6f5.png)

**Comment by egamma** [comment]
Fixed by 9a924f71f4ba4dcf03412387c21e1ba8eb35113e.
